src/embeddings/embeddings_calculator.cc

```diff
@@ -123,42 +128,76 @@ class EmbeddingsCalculator : public CalculatorBase {
+                            handler.setPromptTokensUsage(it->second.get_size());
+                            continue;
                         }
-                    } else {
-                        for (int i = 0; i < it->second.get_byte_size(); i++) {
-                            attendedTokens += reinterpret_cast<uint8_t*>(it->second.data())[i];
+                        size_t attendedTokens = 0;
+                        if (it->second.get_element_type() == ov::element::Type_t::i64) {
+                            for (int i = 0; i < it->second.get_size(); i++) {
+                                attendedTokens += reinterpret_cast<int64_t*>(it->second.data())[i];
+                            }
+                        } else if (it->second.get_element_type() == ov::element::Type_t::i32) {
+                            for (int i = 0; i < it->second.get_size(); i++) {
+                                attendedTokens += reinterpret_cast<int32_t*>(it->second.data())[i];
+                            }
+                        } else {
+                            for (int i = 0; i < it->second.get_byte_size(); i++) {
+                                attendedTokens += reinterpret_cast<uint8_t*>(it->second.data())[i];
+                            }
                         }
+                        handler.setPromptTokensUsage(attendedTokens);
                     }
-                    handler.setPromptTokensUsage(attendedTokens);
+                }
+            } else if (auto ints = std::get_if<std::vector<std::vector<int64_t>>>(&input)) {
+                size_t token_count_of_longest_document = 0;
+                int64_t pad_token = modelConfig["pad_token_id"].as<int64_t>();
+                size_t tokens = 0;
+                received_batch_size = ints->size();
+                for (auto i : *ints) {
+                    token_count_of_longest_document = std::max(token_count_of_longest_document, i.size());
+                    tokens += i.size();
+                }
+                handler.setPromptTokensUsage(tokens);
+                received_batch_size = ints->size();
+                embeddingsInputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME] = ov::Tensor{
+                    ov::element::i64,
+                    ov::Shape{received_batch_size, token_count_of_longest_document}};
+                embeddingsInputMap[EMBEDDINGS_MODEL_ATTENTION_MASK_NAME] = ov::Tensor{
+                    ov::element::i64,
+                    ov::Shape{received_batch_size, token_count_of_longest_document}};
+                for (size_t i = 0; i < received_batch_size; i++) {
+                    int64_t* input_ids_start = reinterpret_cast<int64_t*>(embeddingsInputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].data()) + i * token_count_of_longest_document;
+                    std::fill(input_ids_start, input_ids_start + token_count_of_longest_document, pad_token);
+                    std::copy(ints->at(i).data(), ints->at(i).data() + ints->at(i).size(), input_ids_start);
```

Comment:
Is `ints` size validated to be equal to `received_batch_size` before hand? If so, you can use `operator[]`. If not, it is good practice to use try catch for throwing functions